Automattic/jetpack
# Protect: Add Scan_History class

## Description
Adds a `Scan_History` class responsible for retrieving scan history data from the `wpcom` endpoint (with caching). Adds the normalized data to the initial state for easy retrieval.

### Other information:

## Jetpack product discussion


- https://github.com/Automattic/jetpack-scan-team/issues/1225

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Checkout this branch
* Fire up your Jurassic Tube
* Enable Protect with a paid upgrade and load `jetpackProtectInitialState` in the browser console
* Verify that the data is accurate and comparable to results retrieved from the `/sites/<site_id>/scan/history` endpoint.
* Remove your upgrade and verify that the `scanHistory` initial state variable is set to `false`
* Ensure no regressions are introduced in current functionality
